src/fprime_gds/executables/data_product_writer.py

```diff
@@ -0,0 +1,755 @@
+#
+# Where:
+# <binFile> is the path to the binary file generated by the F' flight software that contains 
+# the data product to be deserialized and written to a JSON file.
+#
+# In addition to the binary file, the program assumes that two essential JSON files are 
+# present in the same directory as the program:
+# 1. dictionary.json - The JSON dictionary file that defines the structure and types of data 
+#    contained in the binary file.
+# 2. dpspec.json - The data product specification file that defines the format of the data 
+#    product's header.
+#
+# These files must be present in the working directory for the program to function correctly, 
+# as they provide the necessary context for decoding the binary data and ensuring the 
+# integrity of the output.
+#
+# The program does not require any additional command-line options or flags for its basic 
+# operation. Once executed with the correct binary file path and with the necessary JSON 
+# files in place, it will perform the series of steps outlined previously, culminating in 
+# the generation of a JSON file with the same base name as the binary file but with a .json 
+# extension.
+#
+# ------------------------------------------------------------------------------------------
+
+import struct
+import json
+import os
+import sys
+from typing import List, Dict, Union, ForwardRef
+from pydantic import BaseModel, field_validator
+from typing import List, Union
+import argparse
+from binascii import crc32
+
+class bcolors:
+    HEADER = '\033[95m'
+    OKBLUE = '\033[94m'
+    OKGREEN = '\033[92m'
+    WARNING = '\033[93m'
+    FAIL = '\033[91m'
+    ENDC = '\033[0m'
+    BOLD = '\033[1m'
+    UNDERLINE = '\033[4m'
+
+
+DP_HEADER_FILE = "dpspec.json"
+
+# Deserialize the binary file big endian
+BIG_ENDIAN = ">"
+
+# -------------------------------------------------------------------------------------
+# These are common Pydantic classes that 
+# are used by both the dictionary json and the data product json 
+#
+# -------------------------------------------------------------------------------------
+
+class IntegerType(BaseModel):
+    name: str
+    kind: str
+    size: int
+    signed: bool
+
+    @field_validator('kind')
+    def kind_qualifiedIdentifier(cls, v):
+        if v != "integer":
+            raise ValueError('Check the "kind" field')
+        return v
+
+class FloatType(BaseModel):
+    name: str
+    kind: str
+    size: int
+
+    @field_validator('kind')
+    def kind_qualifiedIdentifier(cls, v):
+        if v != "float":
+            raise ValueError('Check the "kind" field')
+        return v
+
+class BoolType(BaseModel):
+    name: str
+    kind: str
+    size: int
+
+    @field_validator('kind')
+    def kind_qualifiedIdentifier(cls, v):
+        if v != "bool":
+            raise ValueError('Check the "kind" field')
+        return v
+
+Type = ForwardRef('Type')
+ArrayType = ForwardRef('ArrayType')
+EnumType = ForwardRef('Enumtype')
```

Comment:
I'm not familiar with what this does, but does capitalization matter here? `EnumType` instead of `Enumtype` ?